### PR TITLE
add OperatorGroup reconciliation

### LINKED ISSUE #2516: e2e InstallPlan failed on installplan_e2e_test.go:3242 "should clear clear up the condition in the InstallPlan status that contains an error message when a valid OperatorGroup is created"
## Bug Report

```
Running Suite: End-to-end
=========================
Random Seed: 1639156900
Will run 1 of 169 specs

SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
Install Plan when an InstallPlan is created with no valid OperatorGroup present 
  should clear clear up the condition in the InstallPlan status that contains an error message when a valid OperatorGroup is created
  /Users/akihikokuroda/development/go/src/operator-lifecycle-manager/test/e2e/installplan_e2e_test.go:3193
[BeforeEach] when an InstallPlan is created with no valid OperatorGroup present
  /Users/akihikokuroda/development/go/src/operator-lifecycle-manager/test/e2e/installplan_e2e_test.go:3167
[It] should clear clear up the condition in the InstallPlan status that contains an error message when a valid OperatorGroup is created
  /Users/akihikokuroda/development/go/src/operator-lifecycle-manager/test/e2e/installplan_e2e_test.go:3193
installplan ip is in phase Installing
Waiting for OperatorGroup(og) to be synced with status.namespaces: []
Waiting for OperatorGroup(og) to be synced with status.namespaces: [ns-s2qdg]
installplan ip is in phase Installing
installplan ip is in phase Installing
.........
installplan ip is in phase Installing
installplan ip is in phase Installing
installplan ip is in phase Installing
[AfterEach] when an InstallPlan is created with no valid OperatorGroup present
  /Users/akihikokuroda/development/go/src/operator-lifecycle-manager/test/e2e/installplan_e2e_test.go:3245
[AfterEach] Install Plan
  /Users/akihikokuroda/development/go/src/operator-lifecycle-manager/test/e2e/installplan_e2e_test.go:55
cleaning up ephemeral test resources...
deleting test subscriptions...
deleting test installplans...
deleting test catalogsources...
deleting test crds...
deleting test csvs...
test resources deleted

• Failure [61.482 seconds]
Install Plan
/Users/akihikokuroda/development/go/src/operator-lifecycle-manager/test/e2e/installplan_e2e_test.go:54
  when an InstallPlan is created with no valid OperatorGroup present
  /Users/akihikokuroda/development/go/src/operator-lifecycle-manager/test/e2e/installplan_e2e_test.go:3159
    should clear clear up the condition in the InstallPlan status that contains an error message when a valid OperatorGroup is created [It]
    /Users/akihikokuroda/development/go/src/operator-lifecycle-manager/test/e2e/installplan_e2e_test.go:3193

    Timed out after 60.004s.
    Expected
        <bool>: false
    to be true

    /Users/akihikokuroda/development/go/src/operator-lifecycle-manager/test/e2e/installplan_e2e_test.go:3242
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Summarizing 1 Failure:

[Fail] Install Plan when an InstallPlan is created with no valid OperatorGroup present [It] should clear clear up the condition in the InstallPlan status that contains an error message when a valid OperatorGroup is created 
/Users/akihikokuroda/development/go/src/operator-lifecycle-manager/test/e2e/installplan_e2e_test.go:3242

Ran 1 of 169 Specs in 61.594 seconds
FAIL! -- 0 Passed | 1 Failed | 0 Pending | 168 Skipped
--- FAIL: TestEndToEnd (61.65s)
FAIL

Ginkgo ran 1 suite in 1m13.816506417s
Test Suite Failed
exit status 1
```
**What did you do?**
While I debug https://github.com/operator-framework/operator-lifecycle-manager/issues/2509, this error sometimes happened.  
This error happens more often by making the following code change:
```
diff --git a/pkg/lib/queueinformer/queueinformer_operator.go b/pkg/lib/queueinformer/queueinformer_operator.go
index e009fe221..0cad1738c 100644
--- a/pkg/lib/queueinformer/queueinformer_operator.go
+++ b/pkg/lib/queueinformer/queueinformer_operator.go
@@ -285,7 +285,7 @@ func (o *operator) processNextWorkItem(ctx context.Context, loop *QueueInformer)
 
        // Sync and requeue on error (throw out failed deletion syncs)
        err := loop.Sync(ctx, event)
-       if requeues := queue.NumRequeues(item); err != nil && requeues < 8 && event.Type() != kubestate.ResourceDeleted {
+       if requeues := queue.NumRequeues(item); err != nil && requeues < 5 && event.Type() != kubestate.ResourceDeleted {
                logger.WithField("requeues", requeues).Trace("requeuing with rate limiting")
                utilruntime.HandleError(errors.Wrap(err, fmt.Sprintf("sync %q failed", item)))
                queue.AddRateLimited(item)
```

**What did you expect to see?**
No error 

**What did you see instead? Under which circumstances?**
Error above

**Environment**
* operator-lifecycle-manager version:

* Kubernetes version information:

* Kubernetes cluster kind: 

**Possible Solution**

**Additional context**

### PR DESCRIPTION
**Pull request by akihikokuroda**
Signed-off-by: akihikokuroda <akihikokuroda2020@gmail.com>

**Description of the change:**
Add OperatorGroup reconciliation in Catalog operator so that the operator group creation can trigger InstallPlan sync.

**Motivation for the change:**
Closes #2516
**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

**Comment by perdasilva** [comment]
Hey @akihikokuroda thanks for this! Would it be possible to add unit tests at all?

**Comment by akihikokuroda** [comment]
@perdasilva Thanks for comment.  I'll look into what unit test needs for this change.

**Comment by akihikokuroda** [comment]
I added a small unit test for operatorgroup sync.

File: pkg/controller/operators/catalog/operator.go [hunk 1]
```diff
@@ -526,10 +547,20 @@ func (o *Operator) syncObject(obj interface{}) (syncError error) {
 	}
 
 	o.requeueOwners(metaObj)
-
 	return o.triggerInstallPlanRetry(obj)
 }
 
+func (o *Operator) syncOperatorGroup(obj interface{}) (syncError error) {
+	// Assert as metav1.Object
```
Comment (awgreene) [inline pkg/controller/operators/catalog/operator.go:554]: Nit - I don't think this comment adds any value.

File: pkg/controller/operators/catalog/installplan_sync.go [hunk 2]
```diff
@@ -74,3 +74,49 @@ func (o *Operator) triggerInstallPlanRetry(obj interface{}) (syncError error) {
...
+	if err != nil {
+		syncError = err
+		return
+	}
+
+	update := func(ip *v1alpha1.InstallPlan) error {
+		out := ip.DeepCopy()
+		out.Status.Phase = v1alpha1.InstallPlanPhaseInstalling
+		now := o.now()
+		out.Status.SetCondition(v1alpha1.ConditionFailed(v1alpha1.InstallPlanInstalled, v1alpha1.InstallPlanReasonComponentFailed, "OperatorGroup updated", &now))
```
Comment (awgreene) [inline pkg/controller/operators/catalog/installplan_sync.go:97]: Is this the right condition to add? If we are re-triggering the installPlan I don't think a component failed anymore.

File: pkg/controller/operators/catalog/operator.go (hunk 1 above)
Comment (akihikokuroda) [reply pkg/controller/operators/catalog/operator.go:554]: Thanks! I'll take it out.

File: pkg/controller/operators/catalog/installplan_sync.go (hunk 2 above)
Comment (akihikokuroda) [reply pkg/controller/operators/catalog/installplan_sync.go:97]: I see.   That's right.  I'll change to ConditionMet.

**Comment by akihikokuroda** [comment]
I saw this panic.  It's probably unrelated the changes in this PR so far but I add some code to prevent this in this PR.  This panic causes the same e2e failure.
```
2022-01-18T18:23:06.480373869Z stderr F time="2022-01-18T18:23:06Z" level=debug msg="Got source event: grpc.SourceState{Key:registry.CatalogKey{Name:\"mock-ocs-main-m48zw\", Namespace:\"operator-lifecycle-manager\"}, State:2}"
2022-01-18T18:23:06.480396869Z stderr F time="2022-01-18T18:23:06Z" level=info msg="state.Key.Namespace=operator-lifecycle-manager state.Key.Name=mock-ocs-main-m48zw state.State=READY"
2022-01-18T18:23:06.490549584Z stderr F time="2022-01-18T18:23:06Z" level=error msg="error updating InstallPlan status" id=ilvfL ip=ip namespace=ns-d8pd9 phase=Installing updateError="Operation cannot be fulfilled on installplans.operators.coreos.com \"ip\": the object h\
as been modified; please apply your changes to the latest version and try again"
2022-01-18T18:23:06.495193536Z stderr F panic: runtime error: invalid memory address or nil pointer dereference
2022-01-18T18:23:06.495203736Z stderr F [signal SIGSEGV: segmentation violation code=0x1 addr=0xf4 pc=0x942671d]
2022-01-18T18:23:06.495206836Z stderr F
2022-01-18T18:23:06.495211136Z stderr F goroutine 433 [running]:
2022-01-18T18:23:06.495315737Z stderr F github.com/operator-framework/operator-lifecycle-manager/pkg/controller/operators/catalog.(*Operator).syncInstallPlans(0xbeae410, {0x98647e0, 0xc412120})
2022-01-18T18:23:06.495387838Z stderr F         /home/runner/work/operator-lifecycle-manager/operator-lifecycle-manager/pkg/controller/operators/catalog/operator.go:1593 +0x9ed
2022-01-18T18:23:06.495486339Z stderr F github.com/operator-framework/operator-lifecycle-manager/pkg/lib/queueinformer.LegacySyncHandler.ToSyncerWithDelete.func1({0x9b5e758, 0xbc7a150}, {0x9b55b6c, 0xbfdb240})
2022-01-18T18:23:06.495491739Z stderr F         /home/runner/work/operator-lifecycle-manager/operator-lifecycle-manager/pkg/lib/queueinformer/queueinformer.go:181 +0xed
2022-01-18T18:23:06.495630841Z stderr F github.com/operator-framework/operator-lifecycle-manager/pkg/lib/kubestate.SyncFunc.Sync(0xbc9e7f0, {0x9b5e758, 0xbc7a150}, {0x9b55b6c, 0xbfdb240})
2022-01-18T18:23:06.495714042Z stderr F         /home/runner/work/operator-lifecycle-manager/operator-lifecycle-manager/pkg/lib/kubestate/kubestate.go:184 +0x3c
2022-01-18T18:23:06.495779942Z stderr F github.com/operator-framework/operator-lifecycle-manager/pkg/lib/queueinformer.(*QueueInformer).Sync(...)
2022-01-18T18:23:06.495784942Z stderr F         /home/runner/work/operator-lifecycle-manager/operator-lifecycle-manager/pkg/lib/queueinformer/queueinformer.go:35
2022-01-18T18:23:06.495876443Z stderr F github.com/operator-framework/operator-lifecycle-manager/pkg/lib/queueinformer.(*operator).processNextWorkItem(0xbe6ad90, {0x9b5e758, 0xbc7a150}, 0xbca79e0)
2022-01-18T18:23:06.495919944Z stderr F         /home/runner/work/operator-lifecycle-manager/operator-lifecycle-manager/pkg/lib/queueinformer/queueinformer_operator.go:287 +0x887
2022-01-18T18:23:06.496032045Z stderr F github.com/operator-framework/operator-lifecycle-manager/pkg/lib/queueinformer.(*operator).worker(0xbe6ad90, {0x9b5e758, 0xbc7a150}, 0xbca79e0)
2022-01-18T18:23:06.496135646Z stderr F         /home/runner/work/operator-lifecycle-manager/operator-lifecycle-manager/pkg/lib/queueinformer/queueinformer_operator.go:231 +0x39
2022-01-18T18:23:06.496140746Z stderr F created by github.com/operator-framework/operator-lifecycle-manager/pkg/lib/queueinformer.(*operator).start
2022-01-18T18:23:06.496233648Z stderr F         /home/runner/work/operator-lifecycle-manager/operator-lifecycle-manager/pkg/lib/queueinformer/queueinformer_operator.go:221 +0x560
```

**Comment by perdasilva** [comment]
@akihikokuroda could you please resolve conflicts?

**Comment by perdasilva** [comment]
closing PR as stale. Please re-open if it's still important.

**Pull request closed without merging**